src/lib/marshal/guide/processor.isomorphic.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * IMPORTANT:
+ *
+ * This file is suffixed with `.isomorphic` because the code in this file is
+ * meant to run not just in a nodejs environment but also in a browser. For this
+ * reason there are some restrictions for which nodejs imports are allowed in
+ * this module. See `.eslintrc.json` for more details.
+ */
+import { cloneDeep, get, has, set, unset } from "lodash";
+
+import { AnyObj } from "@/lib/helpers/object.isomorphic";
+import { ObjKeyOrArrayIdx, ObjPath } from "@/lib/helpers/object.isomorphic";
+import { FILEPATH_MARKER } from "@/lib/marshal/shared/const.isomorphic";
+import { ExtractionSettings, WithAnnotation } from "@/lib/marshal/shared/types";
+
+import { prepareResourceJson } from "../shared/helpers.isomorphic";
+import { GuideData } from "./types";
+
+export const GUIDE_JSON = "guide.json";
+
+export type GuideDirBundle = {
+  [relpath: string]: string;
+};
+
+/*
+ * Traverse the given guide data and compile extraction settings of every
+ * extractable field into a sorted map.
+ *
+ * NOTE: Currently we do NOT support content extraction at nested levels for
+ * guides.
+ *
+ * TODO: Abstract this helper and re-use where appropriate.
+ */
+type CompiledExtractionSettings = Map<ObjKeyOrArrayIdx[], ExtractionSettings>;
+
+const compileExtractionSettings = (
+  guide: GuideData<WithAnnotation>,
+): CompiledExtractionSettings => {
+  const extractableFields = get(
+    guide,
+    ["__annotation", "extractable_fields"],
+    {},
+  );
+  const map: CompiledExtractionSettings = new Map();
+
+  for (const [key] of Object.entries(guide)) {
+    // If the field we are on is extractable, then add its extraction settings
+    // to the map with the current object path.
+    if (key in extractableFields) {
+      map.set([key], extractableFields[key]);
+    }
+  }
+
+  return map;
+};
+
+/*
+ * For a given guide payload, this function builds a guide "directory bundle".
+ */
+export const buildGuideDirBundle = (
```

Comment:
Pretty much exactly like the equivalent helper for other resources. I ended up going with not extracting `steps` in the extraction settings for guides in marshal, so some of the code here for template extraction is a bit moot but I left it as is given no harm.